docs/resources/stripe_meter.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: "stripe"
+page_title: "Stripe: stripe_meter"
+description: |-
+  The Stripe Meter can be created, modified and configured by this resource.
+---
+
+# stripe_meter
+
+With this resource, you can create a billing meter - [Stripe API billing meter documentation](https://docs.stripe.com/api/billing/meter).
+
+A billing meter is a resource that allows you to track usage of a particular event. For example, you might create a billing meter to track the number of API calls made by a particular user. You can then attach the billing meter to a price and attach the price to a subscription to charge the user for the number of API calls they make.
+
+Related guide: [Usage based billing](https://docs.stripe.com/billing/subscriptions/usage-based)
+
+
+## Example Usage
+
+```hcl
+// meter
+resource "stripe_meter" "sample_meter" {
+  display_name = "A Sample meter"
+  event_name   = "sample_meter"
+
+  customer_mapping {
+    event_payload_key = "stripe_customer_id"
+    type              = "by_id"
+  }
+
+  default_aggregation {
+    formula = "sum"
+  }
+
+  value_settings {
+    event_payload_key = "value"
+  }
+}
+
+
+// tiered price for the product using the meter above
+resource "stripe_price" "price" {
+  // product needs to be defined
+  product        = stripe_product.product.id
+  currency       = "aud"
+  billing_scheme = "tiered"
+  tiers_mode     = "graduated"
+
+  # free up to ten
+  tiers {
+    up_to       = 10
+    unit_amount = 0 // can be omitted
+  }
+
+  tiers {
+    up_to       = 100
+    unit_amount = 300
+  }
+
+  tiers {
+    up_to               = -1
+    unit_amount_decimal = 100.5
+  }
+
+  recurring {
+    interval        = "week"
+    interval_count  = 2
+    usage_type      = "metered"
+    meter          = stripe_meter.sample_meter.id
+  }
+}
+
+```
+
+## Argument Reference
+
+Arguments accepted by this resource include:
+
+* `display_name` - (Required) String. The display name of the meter.
+* `event_name` - (Required) String. The name of the meter event to record usage for. Corresponds with the `event_name` field on meter events.
+* `event_time_window` - (Optional) String. The time window to pre-aggregate meter events for, if any. Possible values are:
+ - `day` - Events are pre-aggregated in daily buckets
+ - `hour` - Events are pre-aggregated in hourly buckets
+
```

Comment:
Can you list all items here and then go into the details later like it is in other resources?

Reply:
For instance here: https://registry.terraform.io/providers/lukasaron/stripe/latest/docs/resources/stripe_portal_configuration

Reply:
The docs are basically copy&paste from stripe's API documentation which I thought was what you were doing everywhere else but I am ok deviating from that and writing custom docs if that's what you want.

Reply:
Yes, I prefer firstly inform user about all available properties and indicate which ones are required. Afterwards you can dig deeper into individual section if the property has more parameters. This is similar in across Terraform documentation.

Reply:
done I think. I wasn't 100% sure if that's what you meant